TwitterMining/tweet_analyzer.py

```diff
@@ -0,0 +1,44 @@
+import re
+
+from load_tweets import TweetExtractor
+
+
+class TweetClassifier:
+
+    def __init__(self, search_query, tweet_count, data_source):
+        self.search_query = search_query
+        self.tweet_count = tweet_count
+        self.data_source = data_source
+
+    def retrieve_tweets(self):
+        tweet_extractor = TweetExtractor(self.data_source)
+        tweets = tweet_extractor.get_tweets()
+        required_tweets = []
+        tweet_index = 0
+
+        for tweet in tweets:
+            if self.search_query in tweet and \
+                    tweet_index <= self.tweet_count:
+
+                required_tweets.append(tweet)
+                tweet_index = tweet_index + 1
+
+        return required_tweets
+
+    def classify_tweets(self):
+        tweets = self.retrieve_tweets()
+        classification = ""
+        for tweet in tweets:
+            if re.search("#+", str(tweet)):
+                classification += "HashTag "
```

Comment:
Why are we appending and not assigning?
Appending makes code error prone.

Reply:
it's because of a logical decision I made that a tweet can be classified into more then one type, but I understand now that if we save the tweets into a data structure there is no need to concatenate or even use a string, so I'll do that